src/concrete/ml/pytest/utils.py

```diff
@@ -333,36 +333,31 @@ def get_sklearn_all_models_and_datasets(
     Returns:
         List: The pytest parameters to use for testing all models available in Concrete ML.
     """
-    return (
-        get_sklearn_linear_models_and_datasets(
+    all_models_and_datasets = []
+    model_classes = set()
+
+    for get_models_and_datasets_func in [
+        get_sklearn_linear_models_and_datasets,
+        get_sklearn_tree_models_and_datasets,
+        get_sklearn_neural_net_models_and_datasets,
+        get_sklearn_neighbors_models_and_datasets,
+    ]:
+        models_and_datasets = get_models_and_datasets_func(
             regressor=regressor,
             classifier=classifier,
             unique_models=unique_models,
             select=select,
             ignore=ignore,
         )
-        + get_sklearn_tree_models_and_datasets(
-            regressor=regressor,
-            classifier=classifier,
-            unique_models=unique_models,
-            select=select,
-            ignore=ignore,
-        )
-        + get_sklearn_neural_net_models_and_datasets(
-            regressor=regressor,
-            classifier=classifier,
-            unique_models=unique_models,
-            select=select,
-            ignore=ignore,
-        )
-        + get_sklearn_neighbors_models_and_datasets(
-            regressor=regressor,
-            classifier=classifier,
-            unique_models=unique_models,
-            select=select,
-            ignore=ignore,
-        )
-    )
+        for (model, dataset_params), _, _ in models_and_datasets:
```

Comment:
this is wrong, we should not do this : we have models where we intentionally have several tests for them (using different parameters: tweedie, classifiers for binary or multi classif, ect)

if you want a single test per model, you should use `UNIQUE_MODELS_AND_DATASETS` instead of `MODELS_AND_DATASETS` 

and if you need to call a specific type of model (like trees for example), then you should use the `unique_models` parameters like `get_sklearn_tree_models_and_datasets(unique_models=True)`

Reply:
> if you want a single test per model, you should use UNIQUE_MODELS_AND_DATASETS instead of MODELS_AND_DATASETS (basically in this PR you make MODELS_AND_DATASETS the same as UNIQUE_MODELS_AND_DATASETS iiuc)

Yep that's not working this is the point of this fix. We are currently checking multiple times each model class. But I went too fast here I didn't make the change at the right place it seems. Thanks will correct.

Reply:
Let me check without this. This test is 5min so it was a 2m win maybe not the most relevant.

Reply:
not sure what you mean by "it's not working" 🤔

Reply:
This is what I see. Here it's for random forest but every model is being tested twice. That's not good.

```
| Folder        | File                   | Test               | Test Options                  |   Duration |
|:--------------|:-----------------------|:-------------------|:------------------------------|-----------:|
| tests/sklearn | test_sklearn_models.py | test_input_support | torch-RandomForestClassifier  |      7.901 |
| tests/sklearn | test_sklearn_models.py | test_input_support | numpy-RandomForestClassifier  |      7.901 |
| tests/sklearn | test_sklearn_models.py | test_input_support | list-RandomForestClassifier   |      8.289 |
| tests/sklearn | test_sklearn_models.py | test_input_support | pandas-RandomForestClassifier |      7.877 |
| tests/sklearn | test_sklearn_models.py | test_input_support | pandas-RandomForestClassifier |      2.314 |
| tests/sklearn | test_sklearn_models.py | test_input_support | list-RandomForestClassifier   |      2.261 |
| tests/sklearn | test_sklearn_models.py | test_input_support | numpy-RandomForestClassifier  |      2.313 |
| tests/sklearn | test_sklearn_models.py | test_input_support | torch-RandomForestClassifier  |      2.367 |
```

Reply:
are you showing tests of `UNIQUE_MODELS_AND_DATASETS` (unexpected) or `MODELS_AND_DATASETS` (expected) ? @jfrery